### PR TITLE
Use package dependencies to manage TypeScript, instead of `$HOME/.dts`

### PR DESCRIPTION
**Pull request by jakebailey**
Rather than `npm install`-ing into `$HOME/.dts`, instead directly depend on all of the typescript versions we support. We already need them all to do anything, so it's no extra work. Now when we drop old versions, they will disappear from user's machines, and we'll no longer clutter `$HOME/.dts`.

`$HOME/.dts` is still used for the suggestions output, though, so we're not all the way there. Would be nice to use a temp dir instead for that.

### CODE REVIEW

**Comment by jakebailey** [comment]
This is now ready for review.

**Review by andrewbranch — APPROVED** [review]
👏

File: tsconfig.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     { "path": "packages/header-parser" },
     { "path": "packages/publisher" },
     { "path": "packages/typescript-versions" },
```
Comment (sandersn) [inline tsconfig.json:12]: this needs to be updated when typescript-versions is updated, right? Can you add that to the instructions in typescript-versions/index.ts?

Comment (jakebailey) [reply tsconfig.json:12]: The list of packages in `package.json` does, yes, though there's a test that enforces that it's up to date so whoever's bumping will definitely know that they forgot. I'll go and add a note, though.

Comment (jakebailey) [reply tsconfig.json:12]: Done.

**Review by sandersn — APPROVED** [review]

**Pull request merged**